nixos/modules/system/activation/top-level.nix

```diff
@@ -353,7 +376,7 @@ in
             "$out/configuration.nix"
         '';
 
-    system.build.toplevel = system;
+    system.build.toplevel = if config.system.includeBuildDependencies then systemWithBuildDeps else system;
```

Comment:
By wrapping the original system derivation, you may cause subtle changes, such as symlinks instead of files, or subtly broken self references that refer to the system without the build references.
If instead of wrapping, we override it, we can avoid such issues.
We can even make that work with the limitation that `closureInfo`, or even the underlying `exportReferencesGraph` feature will not let us use the reference graph of the derivation that requests the graph. 
For example, it could look like this:

```
system = mkDerivation {
  ...
  if [[ -n "''${buildClosureInfo:-}" ]]; then
    ln -s $buildClosureInfo $out/build-closure-info
  fi
  ...
};
...
system.build.toplevel = system.overrideAttrs(oldAttrs: { buildClosureInfo = pkgs.closureInfo { rootPaths = [ system.drvPath ]; } });
```

So here we still have some deception, but here only the input to `closureInfo` is affected, and we can be confident that the build closures for `system` and `system.overrideAttrs ...` are equivalent.

Reply:
Fixed this, that's a much more elegant approach. Thanks!